pyiode/iode_database/equations_database.pyx

```diff
@@ -112,6 +112,7 @@ cdef class Equations(_AbstractDatabase):
             elif isinstance(value, Equation):
                 equation = value
             elif isinstance(value, dict):
+                # GDM> it seems odd to have this code/capability here instead of as an Equation method (e.g. Equation.update(dict) -- I don't know if that also exists)
```

Comment:
I don't really see what is the problem to implement the case where the passed 'value' is a dict in the special method `__setitem__` ? 
For your information, the `__getitem__` and `__setitem__` methods are implemented in the super class AbstractDatabase and calls protected method `_get_object` and `_set_object` which are implemented in each subclass (Comments, ..., Variables).

Reply:
Having some code here is not a problem, but that's a lot of code which work at the level of a single Equation within the Equations class, so I find the implementation should be in an Equation method and here only a call to that method.